corehq/motech/tests/test_repeater_helpers.py

```diff
@@ -0,0 +1,142 @@
+from django.test.testcases import TestCase
+from mock import patch
+from corehq.form_processor.models import CommCareCaseSQL
+from datetime import datetime
+from corehq.form_processor.interfaces.dbaccessors import CaseAccessors
+from corehq.motech.repeater_helpers import get_relevant_case_updates_from_form_json
+
+
+class TestRepeaterHelpers(TestCase):
+
+    def setUp(self):
+        self.domain = 'test-domain'
+        self.extra_fields = []
+        self.form_question_values = {}
+
+        case_1_data = {
+            'case_id': '5ca13e74-8ba3-4d0d-l09j-66371e8895dd',
+            'domain': self.domain,
+            'type': 'paciente',
+            'name': 'case1',
+            'owner_id': 'owner_1',
+            'modified_by': 'modified_by',
+        }
+        case_2_data = {
+            'case_id': '6ca13e74-8ba3-4d0d-l09j-66371e8895dc',
+            'domain': self.domain,
+            'type': 'casa',
+            'name': 'case2',
+            'owner_id': 'owner_2',
+            'modified_by': 'modified_by',
+        }
+
+        self.case_1 = create_commcare_case(case_1_data)
+        self.case_2 = create_commcare_case(case_2_data)
+
+    def tearDown(self):
+        self.case_1.delete()
+        self.case_2.delete()
+
+    @patch.object(CaseAccessors, 'get_cases')
+    def test__get_relevant_case_updates_from_form_json_with_case_types(self, get_cases):
+        get_cases.return_value = [self.case_1, self.case_2]
+
+        result = get_relevant_case_updates_from_form_json(
+            self.domain,
+            _get_form_json(),
+            ['paciente'],
+            self.extra_fields
+        )
+        self.assertEqual(len(result), 2)
```

Comment:
This returns 2 `CaseTriggerInfo` objects referring to the same case, but only one has an actual `update` value. The other one does not `create`, `update` or `close` - it's just there...I don't think this will cause an issue sending it to OpenMRS, since no create, update or other action is specified through this object, but knowing it does nothing and still sending it through does not sit very well with me.

Reply:
I am reluctant to change behaviour for existing integrations. -- Not without reaching out to organizations that would be affected.

But I do agree with your instincts here, and I think we should follow this up.

(Depending on the priority of other backlog items) we should ...
* write tests to show all scenarios that result in these being empty.
* check existing projects to see whether these scenarios apply to them
* make this change on Staging, reach out to organizations that will be affected, and check that it would have no downstream consequences.

It feels like a tiny change, but I think this function is used by both OpenMRS and DHIS2 Tracked Entity integration, and small behaviour changes can have unexpected consequences that we should be sure about.